aries_cloudagent/vc/data_integrity/cryptosuites/eddsa_jcs_2022.py

```diff
@@ -0,0 +1,201 @@
+
+    async def _get_multikey(self, kid: str):
+        """Derive a multikey from the verification method."""
+
+        # If verification method is a did:key URI,
+        # we derive the multikey directly from the value.
+        if kid.startswith("did:key:"):
+            return kid.split("#")[-1]
+
+        # Otherwise we resolve the verification method and extract the multikey.
+        else:
+            verification_method = await DIDResolver().dereference(
+                profile=self.session.profile, did_url=kid
+            )
+            assert (
+                verification_method["type"] == "Multikey"
+            ), "Expected Multikey verificationMethod type"
+
+            return verification_method["publicKeyMultibase"]
+
+    async def verify_proof(self, secured_document: dict):
+        """Verify proof algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#verify-proof-eddsa-jcs-2022.
+        """
+        unsecured_document = secured_document.copy()
+        proof = unsecured_document.pop("proof")
+        proof_options = proof.copy()
+        proof_bytes = multibase.decode(proof_options.pop("proofValue"))
+
+        try:
+            # Currently leaving context processing out of scope,
+            # leaving code commented as it's technically an algorithm step.
+            # Due to the cryptosuite being based on JSON canonicalization,
+            # the integrity of the document is protected without RDF processing.
+
+            # https://www.w3.org/TR/vc-data-integrity/#validating-contexts
+
+            # assert secured_document['@context'] == proof_options['@context']
+            # unsecured_document['@context'] = proof_options['@context']
+
+            transformed_data = self.transformation(unsecured_document, proof_options)
+            proof_config = self.proof_configuration(proof_options)
+            hash_data = self.hashing(transformed_data, proof_config)
+            if not await self.proof_verification(hash_data, proof_bytes, proof_options):
+                raise CryptosuiteError("Invalid signature.")
+
+        except (AssertionError, CryptosuiteError) as err:
+            problem_detail = PROBLEM_DETAILS["PROOF_VERIFICATION_ERROR"] | {
+                "message": str(err)
+            }
+            return {"verified": False, "proof": proof, "problemDetails": [problem_detail]}
+
+        return {"verified": True, "proof": proof, "problemDetails": []}
+
+    async def proof_verification(
+        self, hash_data: bytes, proof_bytes: bytes, options: dict
+    ):
+        """Proof verification algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#proof-verification-eddsa-jcs-2022.
+        """
+        multikey = await self._get_multikey(options["verificationMethod"])
+        return await self.wallet.verify_message(
+            message=hash_data,
+            signature=proof_bytes,
+            from_verkey=self.key_manager._multikey_to_verkey(multikey),
```

Comment:
Similar comment to the sign; I think we should consider updating the wallet to directly accept a multikey value.

Reply:
Agreed, this has been abstracted by the multikeyManager. I think the wallet should just use multikey instead of verkey, since you can derive alot more information from a multikey.

Is this something that should happen in this pr? The manager is carrying this burden for now, we can look at porting some functionalities from the multikey manager to the Key class directly in a subsequent pr

Reply:
verkey seems like an old artifact from indy / Ed25519VerificationKey2018

Reply:
Indeed, definitely an Indy artifact. I'm cool with leaving that update for a future PR.

Reply:
Oh, the one change I would make in this PR is accessing underscored methods. `_multikey_to_verkey` and `_verkey_to_multikey` don't even need to be methods on the MultikeyManager since they aren't accessing self. I think I'd recommend popping those out into just a function in the key manager module that can be reused inside and outside of the manager until we give the wallet the ability to handle multikeys directly.

Reply:
extracted both method as well as `key_type_from_multikey`